### PR TITLE
Try to make fscore more accurate

### PR DESCRIPTION
**Pull request by demonoidv**
Mainly by adding weight depending on the depth.
There also some other little improvement, nothing really game-changing but should make the AI a bit more efficient.

### CODE REVIEW

File: src/algorithm/minimax.rs [hunk 1]
```diff
@@ -6,23 +6,24 @@ use crate::node::Node;
...
-            self.compute_and_set_fscore(&mut candidate);
-            return candidate;
-        }
+        //     return candidate;
+        // }
 
         if maximizing {
-            fscore = Fscore::Value(isize::MIN);
+
+            println!("Maximazing player:\n{}", node.get_item().get_player());
```
Comment (Ragarnoy) [inline src/algorithm/minimax.rs]: Ah bah bravo!

Comment (demonoidv) [reply src/algorithm/minimax.rs]: Yep, just saw that, fixing it

File: src/algorithm/mod.rs [hunk 2]
```diff
@@ -68,33 +68,50 @@ impl Algorithm {
...
+        //     (self.patterns[PatternName::OpenFour], 9999999isize, 99999999isize),
+        //     (self.patterns[PatternName::CloseFour], 1000isize, 99999999isize),
+        //     (self.patterns[PatternName::SplitFourRight], 50isize, 99999999isize),
+        //     (self.patterns[PatternName::SplitFourLeft], 50isize, 99999999isize),
+        //     (self.patterns[PatternName::SplitFourMiddle], 50isize, 99999999isize),
+        //     (self.patterns[PatternName::Five], 10000000000isize, 100000000isize),
+        //     ((0b01100000, 4, true), 500isize, 2000isize),
+        //     ((0b01010000, 5, true), 25isize, 1000isize)
+        // ];
+        // Got score values from https://playgomoku.online/gomoku-offline
```
Comment (Ragarnoy) [inline src/algorithm/mod.rs:97]: Maybe not the greatest idea since it's a no depth game and might be optimized only to play at 0 depth

Comment (demonoidv) [reply src/algorithm/mod.rs:97]: Could be true, but it actually seems to do a better job since I change those scores

**Review by Ragarnoy — APPROVED** [review]
# yes

**Pull request merged**